tests/zfs-tests/tests/functional/cli_root/zfs_snapshot/zfs_snapshot_009_pos.ksh

```diff
@@ -84,9 +84,15 @@ for i in 1 2 3; do
 done
 log_note "verify snapshot contents"
 for ds in $datasets; do
-	status=$($DIRCMP /$ds /$ds/.zfs/snapshot/snap | $GREP "different")
-	[[ -z $status ]] || log_fail "snapshot contents are different from" \
-	    "the filesystem"
+	if is_linux; then 
+		status=$($DIFF -r /$ds /$ds/.zfs/snapshot/snap | $GREP "different")
```

Comment:
This won't work as is. `diff -r` doesn't output the word different when two directory trees different.  However it does return 1 when directories are different and 0 when they are the same.  For example:

```
$ mkdir foo
$ mkdir bar
$ touch foo/a
$ touch foo/b
$ touch bar/a
$ diff -r foo/ bar/; echo $?
Only in foo/: b
1
$ touch bar/b
$ diff -r foo/ bar/; echo $?
0
```